lib/matplotlib/tests/test_png.py

```diff
@@ -46,3 +47,30 @@ def test_imread_png_uint16():
 
     assert (img.dtype == np.uint16)
     assert np.sum(img.flatten()) == 134184960
+
+
+def test_truncated_file():
```

Comment:
Do we really need to test both filenames and streams?  (or rather, there doesn't seem to be a test of successfully reading from a stream (as opposed to a filename), which seems something more valuable to check...)

Reply:
Yes, because they go through different code paths.

Reply:
but not in the erroring part right?  (at that point every possible input has been converted to an object with a read method)

Reply:
:sheep: Yup, I was confused be the `_read_png_data` in the first if / else block and thought that was reading the whole file, but is just pulling off the first 8 bytes.